localgovdrupal/localgov_moderngov_tpl
# ModernGov template page generation module

Bringing in the ModernGov template page module from the Croydon site.

To serve as a ModernGov template, a page needs to met two requirements:
- Uses these tokens in appropriate locations within the page:
  - ```{pagetitle}```
  - ```{breadcrumb}```
  - ```{content}```
  - ```{sidebar}```
- Uses absolute URLs instead of relative URLs.

This new module achieves the above through the use of a dedicated page template and a custom Path processor service.

## How to test
1. Install the localgov_moderngov_tpl module.
2. Open its config form from /admin/config/content/moderngov/settings and enter a *node id* in the only textfield.
3. Clear Drupal cache.
4. Logout and open the node page who's node id was entered in step 2 above.
5. Verify the presence of the ModernGov tokens (e.g. ```{pagetitle}```) in the page.
6. Verify link and asset URLs within the HTML page source.  These should be all absolute URLs.
7. Note that URLs within blocks other than the following blocks will still appear as relative URLs: content block, menu blocks, alert banner block.  This has to be covered in a different ticket.
